content/en/blog/_posts/2023-03-06-statefulset-autodelete.md

```diff
@@ -0,0 +1,102 @@
+    deleted manually.
+
+  * **`whenDeleted` is `Delete` and `whenScaled` is `Retain`.** In this case, PVCs are deleted only when
+    the entire StatefulSet is deleted. If the StatefulSet is scaled down, PVCs are not touched,
+    meaning they are available to be reattached if a scale-up occurs with any data from the previous
+    replica. This might be used for a temporary StatefulSet, such as in a CI instance or ETL
+    pipeline, where the data on the StatefulSet is needed only during the lifetime of the
+    StatefulSet lifetime, but while the task is running the data is not easily reconstructible. Any
+    retained state is needed for any replicas that scale down and then up.
+
+  * **`whenDeleted` and `whenScaled` are both `Delete`.** PVCs are deleted immediately when their
+    replica is no longer needed. Note this does not include when a Pod is deleted and a new version
+    rescheduled, for example when a node is drained and Pods need to migrate elsewhere. The PVC is
+    deleted only when the replica is no longer needed as signified by a scale-down or StatefulSet
+    deletion. This use case is for when data does not need to live beyond the life of its
+    replica. Perhaps the data is easily reconstructable and the cost savings of deleting unused PVCs
+    is more important than quick scale-up, or perhaps that when a new replica is created, any data
+    from a previous replica is not usable and must be reconstructed anyway.
+
+  * **`whenDeleted` is `Retain` and `whenScaled` is `Delete`.** This is similar to the previous case,
+    when there is little benefit to keeping PVCs for fast reuse during scale-up. An example of a
+    situation where you might use this is an Elasticsearch cluster. Typically you would scale that
+    workload up and down to match demand, whilst ensuring a minimum number of replicas (for example:
+    3). When scaling down, data is migrated away from removed replicas and there is no benefit to
+    retaining those PVCs. However, it can be useful to bring the entire Elasticsearch cluster down
+    temporarily for maintenance. If you need to take the Elasticsearch system offline, you can do
+    this by temporarily deleting the StatefulSet, and then bringing the Elasticsearch cluster back
+    by recreating the StatefulSet. The PVCs holding the Elasticsearch data will still exist and the
+    new replicas will automatically use them.
+
+Visit the
+[documentation](/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-policies) to
+see all the details.
+
+## What’s next?
+
+Try it out! The beta Enable the `StatefulSetAutoDeletePVC` feature gate is enabled by default on
+cluster versions 1.27 and greater. Create a StatefulSet using the new policy, test it out and tell
+us what you think!
+
+I'm very curious to see if this owner reference mechanism works well in practice. For example, I
+realized there is no mechanism in Kubernetes for knowing who set a reference, so it’s possible that
```

Comment:
Server Side Apply, no?

Reply:
Hmm that's a good idea. I wish you had reviewed the KEP :-)

I'll have to try that out---even if current controller don't use it, using it here could clean things up.